cpp/src/arrow/util/io_util.cc

```diff
@@ -97,10 +97,32 @@
 #include "arrow/util/logging.h"
 
 // For filename conversion
-#if defined(_WIN32)
+#ifdef _WIN32
 #include "arrow/util/utf8.h"
 #endif
 
+#ifdef _WIN32
+#include <psapi.h>
+#include <windows.h>
+
+#elif __unix__ || __unix || unix || (__APPLE__ && __MACH__)
+#include <sys/resource.h>
+
+#if __APPLE__ && __MACH__
```

Comment:
We use `#if __APPLE__` quite a bit in the code base.  However, we never use `__MACH__`.  Do you know what this is adding that `__APPLE__` doesn't already specify?

Reply:
https://sourceforge.net/p/predef/wiki/OperatingSystems/ here it uses both conditions, but it seems to be ok if I just let 
 `__APPLE__`